### ISSUE TITLE
Ignition on and off event for Tramigo T23

### ISSUE THREAD

**Issue by busyowo** [tananaev/traccar#1771]
Great job you have going in here...

Please we are not able to see ignition on and off status/events on traccar for tramigo T23, How can we achieve this?

Thank you.

**Comment by tananaev** [comment]
Currently Tramigo decoder doesn't support ignition value.

Please provide some HEX message samples for me to check.

**Comment by busyowo** [comment]
Thank you for your response.

Here is a sample HEX: 80001d3cb000b3000101160f032f1466b475e0564176656e7369732053797353657276653a205374617475732c204750533a203931252c2047534d3a203737252c20475052533a20436f6e6e65637465642c20626174746572793a20313030252c207265706f7274733a2049676e6974696f6e20286f6666292c205374617475732028352c322e302c3732302c3330292c20362e34393239382c20332e33343836352c2031393a3038204d6172203920454f46

**Comment by tananaev** [comment]
Tramigo sends data in human-readable format which makes it very hard to parse on the server side. Do you have protocol documentation for your device? I need it because format is very different from what I've seen before.

**Comment by busyowo** [comment]
![ignitiontramigooff](https://cloud.githubusercontent.com/assets/17742764/13669101/8366b9c2-e6c1-11e5-9ffd-fca294b8ef02.png)
![ignitiontramigoon](https://cloud.githubusercontent.com/assets/17742764/13669102/8369faec-e6c1-11e5-8038-ac48febe8b38.png)


As shown in the attached images, tramigo is sending the ignition status as part of the Hex. However traccar is not reporting this in the database. I used an online decoder: http://www.rapidtables.com/convert/number/hex-to-ascii.htm

NB: I do not have the protocol documentation for the device at the moment.

**Comment by tananaev** [comment]
I can see the text, but I need to know all possible formats to be able to decode it.

**Comment by busyowo** [comment]
Hi tananaev,

For this event type (ignition on and off) there are only two formats.

**1) Ignition on: See sample HEX below**

8000853eb000b8000101fcff032f14665a89e2564176656e7369732053797353657276653a2049676e6974696f6e206f6e2064657465637465642c206d6f76696e672c20302e3135206b6d205357206f66204261626120416e696d61736861756e205374726565742d426f64652054686f6d61732053742e2c20537572756c6572652c204c61676f7320436974792c204e472c20362e34383736352c20332e33343735352c2031303a3031204d6172203131202020454f46

8000973eb000b90001012128032f14667794e2564176656e7369732053797353657276653a2049676e6974696f6e206f6e2064657465637465642c2073746f707065642c20302e3134206b6d205357206f66204261626120416e696d61736861756e205374726565742d426f64652054686f6d61732053742e2c20537572756c6572652c204c61676f7320436974792c204e472c20362e34383736372c20332e33343737332c2031303a3438204d6172203131202020454f46

8000b73eb000ad000101fdd2032f1466c9cbe2564176656e7369732053797353657276653a2049676e6974696f6e206f6e2064657465637465642c206d6f76696e672c20302e3131206b6d2045206f6620416c68616a69204d6173686120526f616420466f6f746272696467652c20537572756c6572652c204c61676f7320436974792c204e472c20362e35303031342c20332e33353434332c2031343a3434204d6172203131202020454f46


**2) Ignition off: See sample HEX below**

8000883eb000d3000101b223032f1466fc89e2564176656e7369732053797353657276653a2049676e6974696f6e206f66662064657465637465642c2049676e4f6e506572696f643a2030303a30323a34312c2073746f707065642c20302e3039206b6d205345206f66204a696e616475205072696d617279205363686f6f6c20416465204f6e6974696d6572696e2053742e2c20537572756c6572652c204c61676f7320436974792c204e472c20362e34383639332c20332e33343636302c2031303a3033204d6172203131202020454f46

80009a3eb000d300010109ff032f1466b195e2564176656e7369732053797353657276653a2049676e6974696f6e206f66662064657465637465642c2049676e4f6e506572696f643a2030303a30353a31342c2073746f707065642c20302e3039206b6d205345206f66204a696e616475205072696d617279205363686f6f6c20416465204f6e6974696d6572696e2053742e2c20537572756c6572652c204c61676f7320436974792c204e472c20362e34383639312c20332e33343636322c2031303a3533204d6172203131202020454f46

8000bc3eb000ba000101622c032f1466bacce2564176656e7369732053797353657276653a2049676e6974696f6e206f66662064657465637465642c2049676e4f6e506572696f643a2030303a30343a30302c206d6f76696e672c20617420416b6572656c6520526f61642d4f67756e6c616e612044726976652c20537572756c6572652c204c61676f7320436974792c204e472c20362e35303630332c20332e33353232382c2031343a3438204d6172203131202020454f46


I hope this is the information you require to decode this event.

Thank you

**Comment by tananaev** [comment]
The format doesn't match your previous example from here: https://github.com/tananaev/traccar/issues/1771#issuecomment-194489156

**Comment by busyowo** [comment]
Please ignore the example you just referenced. It's the format for the status report. For ignition report, the only two valid formats are those ones that I sent to you in my last comment. Please could you process this..
Thank you Sir.

**Comment by tananaev** [comment]
I've added support for ignition.

**Comment by busyowo** [comment]
Thank  you so very much.

Kudos!
